tools/signing.py

```diff
@@ -0,0 +1,87 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+#
+import argparse
+import hashlib
+import os
+import subprocess
+import sys
+
+def parse_args():
+    parser = argparse.ArgumentParser(description='Binary signing tool')
+    parser.add_argument('-m', '--mode', help='Mode (header, sign)')
+    parser.add_argument('-b', '--bin', help='Unsigned binary')
+    parser.add_argument('-o', '--out', help='Output file');
+    parser.add_argument('-p', '--publickey', help='Public key file');
+    parser.add_argument('-s', '--privatekey', help='Private(secret) key file');
+    return parser.parse_args()
+
+def sign_and_write(data, priv_key, out_file):
+    """Signs the data (bytes) with the private key (file path)."""
+    """Save the signed firmware to out_file (file path)."""
+
+    signcmd = [ 'openssl', 'dgst', '-sha256', '-sign', priv_key ]
+    proc = subprocess.Popen(signcmd, stdout=subprocess.PIPE, stdin=subprocess.PIPE, stderr=subprocess.PIPE)
```

Comment:
For people that want to use signed updates, we should declare the OpenSSL dependency somewhere or substitute with a Python library?

Reply:
I'd rather stick w/an OpenSSL dependency.  We ship our own Python3 for the Arduino IDE, and pulling in all the requirements to get a X509 signing module running would be a beast.

How would that requirement be set for P.IO?

Reply:
I wouldn't modify the dependency chain, just forward that responsibility to the user. PIO can also access all the program's in the user's `PATH`, so if they want signed OTA updates, PIO will just call into the same `singing.py` script and it will require `openssl` to be installed. The "standard" user won't turn on signing anyways. I could of course create a OpenSSL package make PlatformIO use it, but unless I get a feature request for that, I wouldn't start doing it.

Reply:
Ah, good.  We'll go with letting the (rare) user who wants to sign make sure he has the tools to do so.  It's in the documentation (not in this PR yet) as a requirement, and you'd generally run `openssl` to generate the public and private key anyways.